src/utils/mod.rs

```diff
@@ -66,7 +68,9 @@ pub fn prefix<'a, I>(iter: I, available_width: usize, delimiter: &str) -> Prefix
 
     // We counted delimiter once too many times,
     // but only if the iterator was non empty.
-    let length = if sum == 0 { sum } else { sum - delimiter_len };
+    let length = sum.saturating_sub(delimiter_len);
+
+    debug_assert!(current_width <= available_width);
 
```

Comment:
The `debug_assert` would have helped to more quickly track down the error. The `saturating_sub` may be faster, is safer (if for some reason 0<sum<delimiter_length, the previous version would panic) and I think it also looks nicer. It already has a good comment, so it's hard not to understand that line anyway.